Production/src/TriggerTools.cc

```diff
@@ -14,14 +18,60 @@ TriggerTools::TriggerTools(EDGetTokenT<edm::TriggerResults>&& _triggerResultsSIM
                            EDGetTokenT<edm::TriggerResults>&& _triggerResultsPAT_token,
                            EDGetTokenT<pat::PackedTriggerPrescales>&& _triggerPrescales_token,
                            EDGetTokenT<pat::TriggerObjectStandAloneCollection>&& _triggerObjects_token,
-                           EDGetTokenT<std::vector<l1extra::L1JetParticle>>&& _l1JetParticles_token) :
+                           EDGetTokenT<std::vector<l1extra::L1JetParticle>>&& _l1JetParticles_token,
+                           std::string _triggerCfg,
+                           analysis::Channel _channel) :
     triggerPrescales_token(_triggerPrescales_token), triggerObjects_token(_triggerObjects_token),
-    l1JetParticles_token(_l1JetParticles_token)
+    l1JetParticles_token(_l1JetParticles_token), triggerCfg(_triggerCfg), channel(_channel)
 {
     triggerResults_tokens[CMSSW_Process::SIM] = _triggerResultsSIM_token;
     triggerResults_tokens[CMSSW_Process::HLT] = _triggerResultsHLT_token;
     triggerResults_tokens[CMSSW_Process::RECO] = _triggerResultsRECO_token;
     triggerResults_tokens[CMSSW_Process::PAT] = _triggerResultsPAT_token;
+
+    ConfigReader config_reader;
+
+    TriggerFileDescriptorCollection trigger_file_descriptors;
+    TriggerFileConfigEntryReader trigger_entry_reader(trigger_file_descriptors);
+    config_reader.AddEntryReader("PATTERN", trigger_entry_reader, true);
+
+    SetupDescriptorCollection setup_descriptors;
+    SetupConfigEntryReader setup_entry_reader(setup_descriptors);
+    config_reader.AddEntryReader("SETUP", setup_entry_reader, false);
+
+    config_reader.ReadConfig(triggerCfg);
+
+    for (const auto& setup : setup_descriptors){
```

Comment:
no need to loop, it should be only one setup per config.
and it is enough deltaPt_map = setup.deltaPt